src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,211 @@
+                            ToDo toDoTask = new ToDo(toDo);
+
+                            this.taskList.addTask(toDoTask);
+
+                            System.out.println("Got it. I've added this task:\n "
+                                    + this.taskList.getTask(this.taskList.getIndex()).toString() + "\nNow you have "
+                                    + (this.taskList.getIndex() + 1) + " tasks in the list.\n");
+
+                            this.storage.writeToFile();
+
+                            this.taskList.incrementIndex();
+                        }
+                    } else if (input.equals("deadline")) {
+                        String deadline = sc.nextLine();
+
+                        if (deadline.equals("")) {
+                            throw new DukeException("☹ OOPS!!! " +
+                                    "The description of a deadline cannot be empty.\n");
+                        } else {
+                            int integer = deadline.indexOf("/by");
+                            String description = deadline.substring(0, integer - 1);
+                            String by = deadline.substring(integer + 4);
+
+                            Deadline deadLineTask = new Deadline(description, by);
+
+                            if (by.contains("/") || by.contains("-")) {
+                                int space = by.indexOf(' ');
+                                String date = by.substring(0, space);
+                                String time = by.substring(space + 1);
+
+                                if (date.contains("/")) {
+                                    int firstSlash = date.indexOf('/', 0);
+                                    int secondSlash = date.indexOf('/', firstSlash + 1);
+                                    String day = date.substring(0, firstSlash);
+                                    String month = date.substring(firstSlash + 1, secondSlash);
+
+                                    if (day.length() == 1) {
+                                        day = "0" + day;
+                                    }
+
+                                    if (month.length() == 1) {
+                                        month = "0" + month;
+                                    }
+
+                                    String year = date.substring(secondSlash + 1);
+                                    date = (year + "-" + month + "-" + day);
+                                }
+
+                                String min = time.substring(2);
+                                String hour = time.substring(0, 2);
+                                time = (hour + ":" + min);
+
+                                deadLineTask.parseTime(time);
+                                deadLineTask.parseDate(date);
+                            }
+
+                            this.taskList.addTask(deadLineTask);
+
+                            System.out.println("Got it. I've added this task:\n "
```

Comment:
You can have a class that handles all the header and footer of the command, so that you wont have to type it out for every single task type.